portal/get_patient_info.php

```diff
@@ -66,27 +66,48 @@
 $logit = new ApplicationTable();
 require_once("$srcdir/authentication/common_operations.php");
 require_once("$srcdir/user.inc");
-$password_update = isset($_SESSION['password_update']);
+$password_update = isset($_SESSION['password_update']) ? $_SESSION['password_update'] : 0;
 unset($_SESSION['password_update']);
 $plain_code = $_POST['pass'];
 
+use OpenEMR\Common\Csrf\CsrfUtils;
+
 $authorizedPortal = false; // flag
 DEFINE("TBL_PAT_ACC_ON", "patient_access_onsite");
 DEFINE("COL_PID", "pid");
 DEFINE("COL_POR_PWD", "portal_pwd");
 DEFINE("COL_POR_USER", "portal_username");
+DEFINE("COL_POR_LOGINUSER", "portal_login_username");
 DEFINE("COL_POR_SALT", "portal_salt");
 DEFINE("COL_POR_PWD_STAT", "portal_pwd_status");
+
+// This is now an account id. Legacy username. Will force case sensitive regardless of collation
 $sql = "SELECT " . implode(",", array(
-    COL_ID,
-    COL_PID,
-    COL_POR_PWD,
-    COL_POR_SALT,
-    COL_POR_PWD_STAT
-)) . " FROM " . TBL_PAT_ACC_ON . " WHERE " . COL_POR_USER . "=?";
+        COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+    " WHERE BINARY " . COL_POR_USER . "= ?";
 $auth = privQuery($sql, array(
     $_POST['uname']
 ));
+if ($password_update === 2) {
+    $auth[COL_POR_LOGINUSER] = '';
+}
+// real username
+if ($auth !== false && !empty($auth[COL_POR_LOGINUSER])) {
+    $sql = "SELECT " . implode(",", array(
+            COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+        " WHERE BINARY " . COL_POR_LOGINUSER . "= ?";
+    $auth = privQuery($sql, array(
+        $_POST['uname']
+    ));
+} elseif ($auth === false) {
+    $sql = "SELECT " . implode(",", array(
+            COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+        " WHERE BINARY " . COL_POR_LOGINUSER . "= ?";
+    $auth = privQuery($sql, array(
+        $_POST['uname']
+    ));
```

Comment:
Am I crazy or is the code the exact same in the 2 code blocks above?

Reply:
Also couldn't some serious weirdness happen if a user happens to have a COL_POR_USER that is the same as another user's COL_POR_LOGINUSER? I think I know of a good solution for this if this is indeed a problem.